xarray/tests/test_backends.py

```diff
@@ -3604,6 +3604,8 @@ def test_open_mfdataset(self) -> None:
                 ) as actual:
                     assert isinstance(actual.foo.variable.data, da.Array)
                     assert actual.foo.variable.data.chunks == ((5, 5),)
+                    print(original)
+                    print(actual)
                     assert_identical(original, actual)
```

Comment:
There is some counter-intuitive stuff going on in this test that I wanted to note.

1) The test asserts that a Dataset contains chunked variables, but is also "identical" to an in-memory array.
2) The test nonetheless passes on `main`!
3) At one point in my PR's development, this assert was raising, but misleading me as to why. The failing assert returned
```python
>   assert_identical(actual, original)
E   AssertionError: Left and right Dataset objects are not identical
E                  
E                
E   Differing data variables:
E       L   foo      (x) float64 dask.array<chunksize=(5,), meta=np.ndarray>
E       R   foo      (x) float64 1.764 0.4002 0.9787 2.241 ... -0.1514 -0.1032 0.4106
```
which makes it seem like it's failing because L is chunked whereas R is in-memory. But (1) and (2) show that's **not actually the reason this assert raised here**. In fact if you add print statements and an `assert False` to this test on `main` you see this output:

```python
<xarray.Dataset>
Dimensions:  (x: 10)
Dimensions without coordinates: x
Data variables:
    foo      (x) float64 1.764 0.4002 0.9787 2.241 ... -0.1514 -0.1032 0.4106
<xarray.Dataset>
Dimensions:  (x: 10)
Dimensions without coordinates: x
Data variables:
    foo      (x) float64 dask.array<chunksize=(5,), meta=np.ndarray>
```

4. It seems the actual reason this assert failed in my case was the the `name` passed to `dask.array.from_array` for chunked variables was different for L and for R. I only found this out by fixing the `name` my code passes down, hence making the test magically pass again.


So am I right in concluding that:
a) `open_mfdataset` does some weird hiding of the fact that it can produce datasets with no `Dataset.chunks`, but whose underlying variables are actually chunked?
b) `assert_identical` will still pass in this counter-intuitive case?
c) `assert_identical` can however fail if dask task names for variables are different, but not tell you that's why it's failing?

Have I understood this correctly @shoyer @dcherian or am I barking up the wrong tree?